hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java

```diff
@@ -78,7 +78,13 @@ public Cacheable getBlock(BlockCacheKey cacheKey, boolean caching,
     // we end up calling l2Cache.getBlock.
     // We are not in a position to exactly look at LRU cache or BC as BlockType may not be getting
     // passed always.
-    return l1Cache.containsBlock(cacheKey)?
+    boolean existInL1 = l1Cache.containsBlock(cacheKey);
+    if (!existInL1 && updateCacheMetrics) {
+      // If the block does not exist in L1, the containsBlock should be counted as one miss.
+      l1Cache.getStats().miss(false, cacheKey.isPrimary(), cacheKey.getBlockType());
+    }
+
+    return existInL1?
```

Comment:
Doesn't the logic become different? The original version will still getBlock from L2, but yours just return existInL1?
I don't get it here.

Reply:
No, the return logic is in 88 & 89 lines.

Reply:
line 88 & 89 are not changed